### PR TITLE
Fluent: Add new color palettes

### PR DESCRIPTION
**Pull request by mikewheaton**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adds new palette slots for Fluent. The existing slots remain in place with the same colors, so that there should be no **design changes by default at this point**. Those who want to opt-in to using the Fluent colors can create a custom theme using the palette:

```
import { FluentPalette } from 'office-ui-fabric-react';

const fluentTheme = createTheme({
  palette: FluentPalette
});

<Fabric>
  <Customizer settings={{theme: fluentTheme}}>
    ... components here ...
  </Customizer>
</Fabric>

```

### CODE REVIEW

**Comment by phkuo** [comment]
@mikewheaton I'll review this more later today, but at a first glance:
- We shouldn't continue having color slots named black/white, or gray.
- Why are there so many shades of red/orange/blue etc colors? Do we need that many reds?? Note that these colors rarely change theme-to-theme, and we hand-tweak each to be accessible against the theme. And these colors are almost never used unless color conveys meaning, such as for errors or warnings.
- Product-specific colors should NOT be in the theme. They can be variables elsewhere because they do not theme. Furthermore, do they belong in Fabric as a platform?
- What are the new versions themePrimary etc? The blue*s?

**Comment by mikewheaton** [comment]
@phkuo Most of these colors wouldn't be altered by a theme, but represent the palette of colors that various themes would be composed of. Should these be made available somewhere else? We need some way to represent the full set of 'crayons' that can be used within a theme, and these names should match the colors (e.g. black) as they are intentionally descriptive and not semantic.

Where would add the product-specific colors? I agree that they should never be themed, I'm just looking for a place to store the palettes.

The 'theme' colors have been replaced with the communication palette. For example, `themePrimary` is now `communication60`.

**Comment by phkuo** [comment]
@mikewheaton why don't we just make them themes then? Just have a file with a JSON blob representing a theme. Or preferably, some code that runs that generates a theme (so it'll be forwards compatible with any new stuff). We could also have a separate file of namedColors (as Office calls them) or or productColors and make it available as JS vars, since we'll never need to use them in SASS. (SASS usage will be limited to existing theme slots which the theme will take care of exposing to the client at runtime.)

File: packages/styling/src/styles/DefaultPalette.ts [hunk 1]
```diff
@@ -24,30 +326,71 @@ export const DefaultPalette: IPalette = {
...
+  orange: '#d83b01',
+  orangeLight: '#ea4300',
+  purpleDark: '#32145a',
+  purple: '#5c2d91',
+  blueDark: '#002050',
+  tealDark: '#004b50',
+  greenDark: '#004b1c',
+};
+
+// Overrides deprecated colors with the new Fluent colors. This allows for Fluent
```
Comment (jordandrako) [inline packages/styling/src/styles/DefaultPalette.ts]: Shouldn't semantic slots be defined in theme.ts under the `_makeSemanticColorsFromPalette` function? Also, consider my changes to deprecation here: https://github.com/OfficeDev/office-ui-fabric-react/pull/4512/files#diff-6118735142300c79db4d510e148e28efR127

File: packages/styling/src/styles/DefaultPalette.ts [hunk 2]
```diff
@@ -2,6 +2,310 @@ import { IPalette } from '../interfaces/index';
...
+  communication80: '#005a9e',
+  communication70: '#106ebe',
+  communication60: '#0078d7',
+  communication50: '#2b88d8',
+  communication40: '#71afe5',
+  communication30: '#c7e0f4',
+  communication20: '#deecf9',
+  communication10: '#eff6fc',
+
+  // Excel
```
Comment (jordandrako) [inline packages/styling/src/styles/DefaultPalette.ts]: We should move these somewhere else. What about making a library of themes in the styling package?

Something like this fake code:

```tsx
import { FluentPalette } from 'office-ui-fabric-react';
import { IExcelTheme } from './ThemeLibrary.types.ts'

export const ExcelTheme = createTheme({
  palette: { 
    ...FluentPalette,
    excel70: '#002901',
    excel60: '#0b4218',
    etc...
  }
});
```

**Pull request closed without merging**